src/main.rs

```diff
@@ -36,31 +46,153 @@ enum OutputFormat {
+
+    let password =
+        rpassword::prompt_password("Please enter a password to encrypt this private key: ")
+            .unwrap();
+
+    let confirmation = rpassword::prompt_password("Please confirm your password: ").unwrap();
+
+    if password != confirmation {
+        println!("Passwords do not match -- try again!");
+        std::process::exit(1);
+    }
+
+    // Wallets are created in ~/.fuel/wallets/ following the format below:
+    // <index>_<public_address>/<uuid>.
+    // The index is the wallet's index in the list of wallets.
+    let path = path.unwrap_or_else(|| {
+        let mut path = home::home_dir().unwrap();
+        path.push(DEFAULT_WALLETS_VAULT_PATH);
+        path.push(format!("{}_{}", get_next_wallet_index(), wallet.address()));
+
+        // create directory if it doesn't exist.
+        if !path.exists() {
+            std::fs::create_dir_all(&path).unwrap();
+        }
+
+        path
+    });
+
+    // Encrypt the wallet and store it in the vault.
+    let uuid = wallet.encrypt(path, password).unwrap();
+
+    // Prints to an alternate screen.
+    // This prevents the mnemonic phrase from being printed to the terminal.
+    let mut screen = AlternateScreen::from(stdout());
+    writeln!(screen, "JSON Wallet uuid: {}\n", uuid).unwrap();
+    writeln!(screen, "Wallet public address: {}\n", wallet.address()).unwrap();
+    writeln!(screen, "Wallet mnemonic phrase: {}\n", mnemonic).unwrap();
+    screen.flush().unwrap();
+
+    let mut input = String::new();
+    println!("### Do not share or lose this mnemonic phrase! Press any key to complete. ###");
+    std::io::stdin().read_line(&mut input).unwrap();
+}
+
+async fn import_wallet(phrase: String) {
+    let wallet = Wallet::new_from_mnemonic_phrase(&phrase, None).unwrap();
+
+    println!("Wallet imported: {}", wallet.address());
 }
 
-fn import_phrase(phrase: String) -> anyhow::Result<Mnemonic> {
-    Mnemonic::from_phrase(&phrase, English)
+// Walks through ~/.fuel/wallets/ and returns the next index based
+// on the number of wallets in the vault.
+fn get_next_wallet_index() -> usize {
+    let mut path = home::home_dir().unwrap();
+    path.push(DEFAULT_WALLETS_VAULT_PATH);
+
+    let dirs = match std::fs::read_dir(path) {
+        Ok(dirs) => dirs,
+        Err(_) => {
+            return 0;
+        }
+    };
+
+    let mut highest_index = 0;
+    for dir in dirs {
+        let dir = dir.unwrap();
+        let path = dir.path();
+
+        if path.is_dir() {
+            let last_component = path.components().last().unwrap();
+            let split: Vec<&str> = last_component
+                .as_os_str()
+                .to_str()
+                .unwrap()
+                .split('_')
+                .collect();
+            let index = split.first().unwrap().parse::<usize>().unwrap();
+            if index > highest_index {
+                highest_index = index;
+            }
+        }
+    }
+
+    highest_index + 1
 }
```

Comment:
A functional alternative, for the `.map` lovers :D 
```suggestion
// Walks through ~/.fuel/wallets/ and returns the next index based
// on the number of wallets in the vault.
fn get_next_wallet_index() -> usize {
    let path = home::home_dir().unwrap().join(DEFAULT_WALLETS_VAULT_PATH);

    let directories = match list_dirs_in(path) {
        Ok(entries) => entries,
        Err(_) => {
            return 0;
        }
    };

    directories
        .into_iter()
        .map(num_of_wallets_in_vault)
        .max()
        .unwrap_or(1)
}

fn num_of_wallets_in_vault<T: AsRef<Path>>(dir: T) -> usize {
    extract_wallet_num_str(dir.as_ref())
        .unwrap()
        .parse()
        .unwrap()
}

fn extract_wallet_num_str(s: &Path) -> Option<&str> {
    s.file_name()?.to_str()?.split('_').next()
}

fn list_dirs_in(path: PathBuf) -> io::Result<Vec<PathBuf>> {
    let entries = std::fs::read_dir(path)?
        .into_iter()
        .collect::<Result<Vec<_>, _>>()?;

    Ok(entries
        .into_iter()
        .map(|entry| entry.path())
        .filter(|path| path.is_dir())
        .collect())
}
```